src/Events/ModuleBaseEvent.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Nwidart\Modules\Events;
+
+use Illuminate\Console\Command;
+use Nwidart\Modules\Module;
+use Nwidart\Modules\Repository;
+
+/**
+ * Class ModuleBaseEvent
+ * @package Nwidart\Modules\Events
+ */
+abstract class ModuleBaseEvent
+{
+    /**
+     * The module instance.
+     *
+     * @var Module
+     */
+    protected $module;
+
+    /**
+     * The console command instance.
+     *
+     * @var Command
+     */
+    protected $console;
+
+    public function __construct($name, Repository $repository, Command $console)
+    {
+        $this->module = $repository->get($name);
```

Comment:
This is not to have the name but the module itself. Should i do it in the installer instead ?